pkg/manager/member/tiflash_upgrader.go

```diff
@@ -101,6 +111,25 @@ func (u *tiflashUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Statefu
 			if store.State != v1alpha1.TiKVStateUp {
 				return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store state is not UP", ns, tcName, podName)
 			}
+
+			needCheckStatus := false
+			tiflashVersion := tc.TiFlashVersion()
+			if _, ok := tiflashVersionsNeedCheckStatus[tiflashVersion]; ok {
+				needCheckStatus = true
+			} else if ver, err := semver.NewVersion(tiflashVersion); err == nil && ver.Compare(tiflashV512) >= 0 { // NOTE: if parse image version failed, will skip this check
+				needCheckStatus = true
+			}
+			if needCheckStatus {
```

Comment:
Can we always check the status for all versions but handle HTTP 404?

Reply:
If do that, when the new version tiflash return HTTP 404, the check will skip, I think it is ambiguous.